targets/TARGET_NXP/TARGET_MCUXpresso_MCUS/TARGET_LPC/flash_api.c

```diff
@@ -80,10 +80,39 @@ int32_t flash_program_page(flash_t *obj, uint32_t address, const uint8_t *data,
 
     status = FLASHIAP_PrepareSectorForWrite(sector_number, sector_number);
     if (status == kStatus_FLASHIAP_Success) {
-        status = FLASHIAP_CopyRamToFlash(address, (uint32_t *)data,
-                                        FSL_FEATURE_SYSCON_FLASH_PAGE_SIZE_BYTES, SystemCoreClock);
-        if (status == kStatus_FLASHIAP_Success) {
-            ret = 0;
+        /* Check if the number of bytes to write is aligned to page size */
+        if (size % FSL_FEATURE_SYSCON_FLASH_PAGE_SIZE_BYTES) {
+            uint8_t page_buffer[FSL_FEATURE_SYSCON_FLASH_PAGE_SIZE_BYTES] = { 0 };
+            uint32_t remaining_bytes = 0;
+
+            /* Find the number of pages and remaining bytes */
+            num_of_bytes = FSL_FEATURE_SYSCON_FLASH_PAGE_SIZE_BYTES * (size / FSL_FEATURE_SYSCON_FLASH_PAGE_SIZE_BYTES);
+            remaining_bytes = (size - num_of_bytes);
+
+            /* Copy the remaining bytes into a temp buffer whose size is page-aligned */
+            memcpy(page_buffer, data + num_of_bytes, remaining_bytes);
+
+            if (num_of_bytes) {
+                /* First write page size aligned bytes of data */
+                status = FLASHIAP_CopyRamToFlash(address, (uint32_t *)data, num_of_bytes, SystemCoreClock);
```

Comment:
The comment for FLASHIAP_CopyRamToFlash() says it can handle num_of_bytes:
_"The addresses should be a 256 byte boundary and the number of bytes should be 256 | 512 | 1024 | 4096."_
Here num_of_bytes can be any number aligned to 256. Its worth double checking whether 256|512|1024|4096 also includes other multiplies of 256... e.g. 768

Reply:
I modified the test to set the test_size to be page_size * 3, the tests passed.

Reply:
I appreciate it - thanks!